src/kernel/src/ucred/mod.rs

```diff
@@ -83,18 +100,31 @@ impl Ucred {
         self.auth.caps.is_unk1() && self.auth.attrs.is_unk2()
     }
 
+    /// See `prison_check` on the PS4 for a reference.
+    pub fn prison_check(&self, other: &Self) -> Result<(), PrisonCheckError> {
```

Comment:
Can we use the other name that more descriptive? `prison_check` is unclear what checks it does.